### PR TITLE
legacy; new 0.24.0-dev version 

### PR DESCRIPTION
**Pull request by tomahawk28**
After 0.23.1 release, introduce new 0.24.0-dev version

### CODE REVIEW

**Comment by xh3b4sd** [comment]
I don't follow what we "roll back" here.

**Comment by tomahawk28** [comment]
Updated the title since it's misleading.

File: pkg/project/project.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package project
 
 var (
-	bundleVersion = "0.23.1"
+	bundleVersion = "0.24.0-dev"
```
Comment (sslavic) [inline pkg/project/project.go]: Already know it will not be a patch, but minor?

Comment (tomahawk28) [reply pkg/project/project.go]: We can back into patch version whenever we need it for next time 😄

Comment (rossf7) [reply pkg/project/project.go]: The docs say it should be a patch version. We upgrade to minor version later if necessary.

https://github.com/giantswarm/giantswarm/blob/master/areas/empowerment/versioning-of-operators/how-to-release-an-operator.md#change-the-operator-version-in-the-code

@tomahawk28 Please do another PR so this is correct.

**Pull request merged**